src/Microsoft.AspNet.Hosting/Program.cs

```diff
@@ -23,31 +24,37 @@ public Program(IServiceProvider serviceProvider)
             _serviceProvider = serviceProvider;
         }
 
-        public void Main(string[] args)
+        public IHostingEngine CreateHost(IConfiguration config)
         {
-            var config = new Configuration();
-            if (File.Exists(HostingIniFile))
-            {
-                config.AddIniFile(HostingIniFile);
-            }
-            config.AddEnvironmentVariables();
-            config.AddCommandLine(args);
-
-            var engine = WebHost.CreateEngine(_serviceProvider, config);
+            var builder = new WebHostBuilder(_serviceProvider);
             var server = config.Get("server");
             if (server != null)
             {
-                engine.UseServer(server);
+                builder.UseServer(server);
             }
             var startup = config.Get("app");
```

Comment:
might be the time to pick new keys /cc @lodejard. server and app are terrible. Lets keep new app and server as a fallback. Maybe something like:

``
var server = config.Get("Hosting:Server") ?? config.Get("server");
var app = config.Get("Hosting:Application") ?? config.Get("app");
``